File: eecs268/2021spring/MWF/2021.04.30.heaps-part-4/MinHeap.cpp
//MinHeap.cpp

template<typename T>
void MinHeap<T>::add(T entry)
{
	if(heapSize == arraySize)
	{
		resize();//doubles current array size
	}
	
	m_arr[heapSize] = entry;
	upheap(heapSize);
	heapSize++;
}

File: eecs268/2025summer/2025.07.16.heap-as-lists-implementation/minheap.py
#minheap.py

class MinHeap:
    def __init__(self):
        self._heap = []

    def add(self, entry):
        self._heap.append(entry)

        self._upheap(len(self._heap)-1)


    def _upheap(self, index):
        #define
